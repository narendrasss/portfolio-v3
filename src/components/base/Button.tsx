import React from 'react';
import Box, { BoxProps } from './Box';

export interface ButtonProps extends BoxProps {
  onClick: React.ChangeEventHandler<HTMLButtonElement>;
  type?: string;
}

const Button: React.FC<ButtonProps> = ({ type = 'button', ...props }) => (
  <Box as="button" data-testid="button" type={type} {...props} />
);

export default Button;
