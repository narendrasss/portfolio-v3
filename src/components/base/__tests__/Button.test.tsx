import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import Button from '../Button';

describe('Button', () => {
  const mockHandler = jest.fn();

  it('renders without crashing', () => {
    const { asFragment } = render(<Button onClick={mockHandler} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls handler when clicked', () => {
    const { getByTestId } = render(<Button onClick={mockHandler} />);
    fireEvent.click(getByTestId('button'));
    expect(mockHandler).toHaveBeenCalled();
  });
});
