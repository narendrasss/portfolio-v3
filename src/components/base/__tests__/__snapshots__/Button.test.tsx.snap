// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without crashing 1`] = `
<DocumentFragment>
  <button
    class="sc-bdVaJa sc-htoDjs qIBAg"
    data-testid="button"
    type="button"
  />
</DocumentFragment>
`;
